### DISCUSSION TITLE
Cannot Add a node

### DISCUSSION THREAD

**Discussion by Dylantje** [zwave-js/zwave-js#6417] [Request Support / Investigate Issue] [answered]
I run:
zwave-js-ui: 8.24.0.30eba25
zwave-js: 11.13.0
home id: 3514694172
home hex: 0xd17dfa1c

In docker with using Domoticz.
I came from open zwave, and doe use the zwave usb: Aeotec Z-Stick Gen 5+
I did use the nodes that where learned on the stick, and they work almost great.

Only when i try to learn a new node toe the stick with the zwave-js web ui option. I cannot learn a node.
When is learn  a node directly to the usb, i can see it in to the webui. Only the node stys dead.. what i try, still dead

Is there a special way to learn nodes?
What can i do wrong?

**Comment by AlCalzone** [comment]
This sounds like you're having range issues - the nodes aren't reachable from where the Z-Stick is located. Maybe you don't have enough repeaters (mains-powered nodes) between the controller and where you're trying to include the nodes?

We also discourage moving around with the stick to include: https://github.com/zwave-js/node-zwave-js/issues/1995
Sometimes there's no alternative though, if the nodes don't support network-wide inclusion, cannot reach the stick directly, and you cannot move them to the stick to include them. In that case, you may have some luck with rebuilding routes to the problematic nodes.

**Reply by Dylantje** [reply]
Thanks... Walking arround the stick was always working in openzwave. Thanks for that.
In open zwave the devices are working always good and i did not have a range issue.

Al my device are power router, i do not use battery devices.
![ScreenShot796](https://github.com/zwave-js/node-zwave-js/assets/5214969/72867ad3-f77a-4f2e-a1fc-4307c7766bcc)

![ScreenShot797](https://github.com/zwave-js/node-zwave-js/assets/5214969/731eb9c0-241a-4b39-a34d-33a90991e132)

The range from the not working device to the firt router device is almost 10 meter

Is there a way to power up the signal ?

And will you please tell me how the correct way is to include a node?

**Reply by AlCalzone** [reply, marked as answer]
> In open zwave the devices are working always good and i did not have a range issue.

That can't be, as the signal is purely hardware-dependent. OpenZWave and Z-Wave JS simply tell the stick to send something to a node - everything that happens then is out of our control.

Your mesh doesn't look very good, lots of routes falling back to the lowest possible speed, I suggest following these guides to try and make it healthier:
1. https://zwave-js.github.io/node-zwave-js/#/troubleshooting/connectivity-issues?id=general-troubleshooting
2. https://zwave-js.github.io/node-zwave-js/#/troubleshooting/network-health?id=testing-the-connection-strength

> And will you please tell me how the correct way is to include a node?

Start inclusion in Z-Wave JS UI, push button on the device.

**Reply by Dylantje** [reply]
Thanks for the suggestions..
Now on my mobile so when i am home i give it a try.

Only how can you see that my nodes fallback to the slowest speed?

When i read well the usb 3 ports are not the best options?? 
Do have also a usb switch between my odroid and the zwave, is that also a great problem?

**Reply by AlCalzone** [reply]
> Only how can you see that my nodes fallback to the slowest speed?

If you open the dropdown at the top of the map, you'll see a legend what the route colors mean.

**Comment by Dylantje** [comment]
THANKS guys
Did learn a lot....
![ScreenShot800](https://github.com/zwave-js/node-zwave-js/assets/5214969/32a33d10-cb16-4712-85f0-69a1a02f18d7)

A lot of work what i did  now..
Now evrything is working fine.
Did set a lot of routes, and buy a extention usb etc...

I hope you say that my network is ok now:-)?
Or is it possible to get every route to a node green?

and what is de the best color for a node?

**Reply by AlCalzone** [reply]
Definitely better!

The node colors only indicate how many hops are needed to get there. Ideally that number is as low as possible.

Whether all routes can be 100 kbps (green) depends on the devices. Some older ones don't support that speed. The red one (9.6 kbps) to the printer is the only one that doesn't have a better alternative. Maybe look into that one if you're having problems with it.
